string.c

```diff
@@ -1493,34 +1493,29 @@ str_replace(VALUE str, VALUE str2)
 static inline VALUE
 str_duplicate(VALUE klass, VALUE str)
 {
-    enum {embed_size = RSTRING_EMBED_LEN_MAX + 1};
     const VALUE flag_mask =
 	RSTRING_NOEMBED | RSTRING_EMBED_LEN_MASK |
 	ENC_CODERANGE_MASK | ENCODING_MASK |
 	FL_TAINT | FL_FREEZE
 	;
     VALUE flags = FL_TEST_RAW(str, flag_mask);
     VALUE dup = str_alloc(klass);
-    MEMCPY(RSTRING(dup)->as.ary, RSTRING(str)->as.ary,
-	   char, embed_size);
+    RSTRING(dup)->as = RSTRING(str)->as;
```

Comment:
You may want to separate these assignments as another PR?

Reply:
Even another commit would be nice, of course.

Reply:
I dropped it since it's fairly unrelated :)